sound/soc/codecs/rt711-sdca.c

```diff
@@ -790,6 +792,59 @@ static int rt711_sdca_fu0f_capture_put(struct snd_kcontrol *kcontrol,
 	return changed;
 }
 
+static int rt711_sdca_ge_select_get(struct snd_kcontrol *kcontrol,
+			      struct snd_ctl_elem_value *ucontrol)
+{
+	return snd_soc_get_enum_double(kcontrol, ucontrol);
+}
+
+static int rt711_sdca_ge_select_put(struct snd_kcontrol *kcontrol,
+			      struct snd_ctl_elem_value *ucontrol)
+{
+	struct soc_enum *e = (struct soc_enum *)kcontrol->private_value;
+	unsigned int *item = ucontrol->value.enumerated.item;
+	struct snd_soc_component *component = snd_soc_kcontrol_component(kcontrol);
+	struct rt711_sdca_priv *rt711 = snd_soc_component_get_drvdata(component);
+	unsigned int val;
+
+	if (item[0] >= e->items)
+		return -EINVAL;
+	val = snd_soc_enum_item_to_val(e, item[0]) << e->shift_l;
+
+	switch (val) {
+	case 0x03:
+		rt711_sdca_index_write(rt711, RT711_VENDOR_REG,
+			RT711_COMBO_JACK_AUTO_CTL1, 0xc689);
+		rt711_sdca_index_write(rt711, RT711_VENDOR_REG,
+			RT711_VREFOUT_CTL, 0x0064);
+		break;
+	case 0x05:
+		rt711_sdca_index_write(rt711, RT711_VENDOR_REG,
+			RT711_COMBO_JACK_AUTO_CTL1, 0xd689);
+		rt711_sdca_index_write(rt711, RT711_VENDOR_REG,
+			RT711_VREFOUT_CTL, 0x1064);
+		break;
+	}
+
+	return snd_soc_put_enum_double(kcontrol, ucontrol);
```

Comment:
still not following. Even if we used a changed in selected_mode as the trigger to update the configuration, the jack_type wouldn't change.

My maybe naive proposal was that the jack detection would be set with the alsa control when the card is created, and whenever an actual jack is detected we would use the value of the alsa control to force the configuration and ignore the detected mode.

Reply:
@plbossart Could we use the module parameter to force the jack type?
I updated the patch. Please take a look.